test/irb/test_history.rb

```diff
@@ -169,15 +182,17 @@ def assert_history(expected_history, initial_irb_history, input)
           f.write(initial_irb_history)
         end
 
-        io = TestInputMethodWithHistory.new
+        io = input_method.new
         io.class::HISTORY.clear
         io.load_history
         if block_given?
-          history = io.class::HISTORY.dup
+          previous_history = []
+          io.class::HISTORY.each { |line| previous_history << line }
           yield IRB.rc_file("_history")
-          io.class::HISTORY.replace(history)
```

Comment:
`Readline::HISTORY` doesn't respond to `replace`.  Updated so that this works for both input methods.